ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/validators.tsx

```diff
@@ -0,0 +1,229 @@
+}
+
+function isUPS(input: string) {
+  try {
+    converter.deserializeUPS(input)
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+function validateUsngGrid(grid: string) {
+  //corner case for ups zone
+  return converter.isUSNG(grid) !== 0 || isUPS(grid)
+}
+
+function gridIsBlank(grid: string) {
+  return grid.length === 0
+}
+
+function dmsPointIsBlank(point: Point) {
+  return point.lat === '' || point.lon === ''
+}
+
+function utmUpsIsBlank(point: UtmUpsPoint) {
+  return point.easting || point.hemisphere || point.northing || point.zoneNumber
+    ? false
+    : true
+}
+
+function validateUtmUpsPoint(point: UtmUpsPoint) {
+  const validation = validateGeo('easting', point)
+  return !validation?.error
+}
+
+function inValidRange(coordinate: DmsCoordinate, maximum: number) {
+  const degrees = parseInt(coordinate.degrees)
+  const minutes = parseInt(coordinate.minutes)
+  const seconds = parseFloat(coordinate.seconds)
+  if (isNaN(seconds)) {
+    return false
+  }
+  if (degrees > maximum || minutes > 60 || seconds > 60) {
+    return false
+  }
+  if (degrees === maximum && (minutes > 0 || seconds > 0)) {
+    return false
+  }
+  return true
+}
+
+function replacePlaceholderWithZeros(numString = '') {
+  while (numString.includes('_')) {
+    if (numString.includes('.')) {
+      numString = numString.replace('_', '0')
+    } else {
+      numString = numString.replace('_', '')
+      numString = '0' + numString
+    }
+  }
+  return numString
+}
+
+function parseDmsCoordinate(coordinate: string) {
+  if (coordinate === undefined) {
+    return undefined
+  }
+
+  const matches = dmsRegex.exec(coordinate)
+  if (!matches) {
+    return undefined
+  }
+  const degrees = replacePlaceholderWithZeros(matches[1])
+  const minutes = replacePlaceholderWithZeros(matches[2])
+  const seconds = replacePlaceholderWithZeros(matches[3])
+  return { degrees, minutes, seconds } as DmsCoordinate
+}
+
+function validateDmsPoint(point: Point) {
+  const latitude = parseDmsCoordinate(point.lat)
+  const longitude = parseDmsCoordinate(point.lon)
+  if (latitude && longitude) {
+    return inValidRange(latitude, 90) && inValidRange(longitude, 180)
+  }
+  return false
+}
+
+function validateDmsLineOrPoly(dms: Point[], type: 'line' | 'polygon') {
+  let defaultValue
+  if (!dms || dms.some(dmsPointIsBlank)) {
+    return { error: true, message: errorMessages.invalidList, defaultValue }
+  }
+  let error = false
+  let message = null
+  switch (type) {
+    case 'line':
+      if (!dms.every(validateDmsPoint)) {
+        error = true
+        message = errorMessages.invalidList
+      } else if (dms.length < 2) {
+        error = true
+        message = errorMessages.tooFewPointsLine
+      }
+      break
+    case 'polygon':
+      if (!dms.every(validateDmsPoint)) {
+        error = true
+        message = errorMessages.invalidList
+      } else if (dms.length < 3) {
+        error = true
+        message = errorMessages.tooFewPointsPolygon
```

Comment:
📝  Something that's a bit curious is that it's really hard to draw a 3 point polygon. 
In almost all cases I tried manually, 3 point polys are invalid at least on the 2D map. 
However, using the 3D map it is a bit easier to get 3 point polys so I kept the existing 3 point poly error message here. 

This code is sliiiightly redundant, but I kept it expanded like this in case any of the types required extra/special validation clauses.